include/spirv-tools/libspirv.h

```diff
@@ -473,6 +473,19 @@ SPIRV_TOOLS_EXPORT const char* spvTargetEnvDescription(spv_target_env env);
 // false and sets *env to SPV_ENV_UNIVERSAL_1_0.
 SPIRV_TOOLS_EXPORT bool spvParseTargetEnv(const char* s, spv_target_env* env);
 
+// Determines the target env value with the least features but which enables
+// the given Vulkan and SPIR-V versions. If such a target is supported, returns
+// true and writes the value to |env|, otherwise returns false.
+//
+// The Vulkan version is given as an unsigned 32-bit number as specified in
+// Vulkan section "29.2.1 Version Numbers": the major version number appears
+// in bits 22 to 21, and the minor version is in bits 12 to 21.  The SPIR-V
+// version is given in the SPIR-V version header word: major version in bits
+// 16 to 23, and minor version in bits 8 to 15.
+SPIRV_TOOLS_EXPORT bool spvParseVulkanEnv(uint32_t vulkan_ver,
```

Comment:
What should glslang do for OpenGL contexts?

Currently, glslang also has paths for getting a context for non-Vulkan contexts.  Keep that code as is?  Or, generalize this to something like (pseudo code):

```
enum clientAPI {
    Vulkan,
    OpenGL
};
SPIRV_TOOLS_EXPORT bool spvParseEnv(clientAPI client, ...
```

Reply:
Yeah, I wondered about OpenGL (and didn't look!).

I can't predict how versions will be described for upcoming APIs. It happens that Vulkan and OpenGL have conventions that fit into a 32-bit int.  I have no idea how WebGPU would fare.  So my preference is to have a separate entry point for each API.  Adding one more for OpenGL is low cost and keeps the parameter descriptions clear.

I'll ruminate over the weekend over this.

Reply:
Yes, separate entry points is better.